lua/fff/rust/lib.rs

```diff
@@ -183,14 +182,7 @@ pub fn wait_for_initial_scan(_: &Lua, timeout_ms: Option<u64>) -> LuaResult<bool
         .as_ref()
         .ok_or_else(|| Error::FilePickerMissing)?;
 
-    let timeout = Duration::from_millis(timeout_ms.unwrap_or(5000)); // Default 5s timeout
-    let start_time = std::time::Instant::now();
-
-    while picker.is_scan_active() && start_time.elapsed() < timeout {
-        std::thread::sleep(Duration::from_millis(50));
-    }
-
-    Ok(!picker.is_scan_active())
+    Ok(picker.wait_for_complete_scan(timeout_ms))
```

Comment:
This makes sense but I think we should just inline this function

Reply:
By inline, are you referring to adding a `#[inline]` attribute or extracting the logic from `wait_for_complete_scan` to `wait_for_initial_scan`?

I'm assuming you're referring to the latter.